spec/index.bs

```diff
@@ -569,6 +561,63 @@ enum GPUBindingType {
 };
 </script>
 
+  * {{GPUBindGroupLayoutBinding/textureDimension}}, {{GPUBindGroupLayoutBinding/multisampled}}:
+    The dimensions and multi-sampling properties of texture bindings.
+
+    Note: This allows Metal-based implementations to back the respective bind
+    groups with `MTLArgumentBuffer` objects that are more efficient to bind at
+    run-time.
+
+  * {{GPUBindGroupLayoutBinding/dynamic}}:
+    For [=uniform-buffer=], [=storage-buffer=], and [=readonly-storage-buffer=] bindings, 
+    indicates that the binding has a dynamic offset. One offset must be passed to 
+    setBindGroup for each dynamic binding in increasing order of
+    {{GPUBindGroupLayoutBinding/binding}} number.
+
+### {{GPUDevice/createBindGroupLayout()|GPUDevice.createBindGroupLayout(GPUBindGroupLayoutDescriptor)}} ### {#GPUDevice-createBindGroupLayout}
+
+A valid {{GPUBindGroupLayout}} object has an <dfn>internal map</dfn> of |u32| <dfn>binding number</dfn>s to internal <dfn>layout binding</dfn> objects. 
+Each [=layout binding=] has a {{GPUShaderStageFlags}} <dfn>visibility</dfn> bitset and a {{GPUBindingType}} <dfn>type</dfn>.
+
+A [=layout binding=] with type {{GPUBindingType/uniform-buffer}}, {{GPUBindingType/storage-buffer}}, or {{GPUBindingType/readonly-storage-buffer}} also has 
+a <dfn>dynamic</dfn> boolean which must be initially <code>false</code>.
+
+A [=layout binding=] with type {{GPUBindingType/sampled-texture}} or {{GPUBindingType/storage-texture}} also has a {{GPUTextureViewDimension}} <dfn>textureDimension</dfn> which must be initially <code>"2d"</code>,
+a {{GPUTextureComponentType}} <dfn>textureComponentType</dfn> which must be initially <code>"float</code>, and a <dfn>multisampled</dfn> boolean which must be initially <code>false</code>.
+
+<div algorithm="GPUDevice.createBindGroupLayout(descriptor)">
+Any time during this algorithm, to <dfn>return with error</dfn> is to:
+    1. Generate a validation error in the current scope with appropriate error message.
+    1. Create a new invalid {{GPUBindGroupLayout}} and return the result.
+
+The <dfn method for "GPUDevice">createBindGroupLayout(|descriptor|)</dfn> method attempts to create a valid {{GPUBindGroupLayout}}.
```

Comment:
I'm not sure if the wording of "attemtps" and "valid" is good here. Saying that a function "attempts" to do something appears to make sense to me if the return type indicates a possible failure. This is not the case here. Also, "valid" is strange because it's hard to imagine what else it could be doing. Seeing "valid" even makes it confusing to me, since it implies that the implementation is trying really hard to create it in such a way that it's "valid", while in fact the implementation doesn't do anything sophisticated.

Reply:
I'm not sure yet how else to describe what to expect in a failure case. We use "invalid" to describe other objects in the API so it stands to reason that the opposite of such would be a "valid" object.

Reply:
The failure cases are described separately as a follow-up: we list the things that are checked and say in which case the result is invalid.
What we shouldn't do is saying that "a method attempts to create a valid XXX", since this implies that the method does more than necessary to make sure it's "valid". We should just say "method creates an XXX" instead.

Reply:
Text was updated to do so. PTAL